src/components/Layout/DrawerNavigation.tsx

```diff
@@ -0,0 +1,196 @@
+import { useLogout } from "../../hooks/useLogout";
+import { AppText } from "./AppText";
+import { size, color, fontSize } from "../../common/styles";
+import { MaterialCommunityIcons } from "@expo/vector-icons";
+import { HelpModalContext } from "../../context/help";
+import { useDrawerContext, DrawerButtons } from "../../context/drawer";
+import Constants from "expo-constants";
+import { APP_BUILD_VERSION } from "../../config";
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    position: "relative"
+  },
+  navLink: {
+    flexDirection: "row",
+    alignItems: "center",
+    paddingHorizontal: size(4),
+    paddingVertical: size(2.5),
+    borderBottomColor: color("grey", 20),
+    borderBottomWidth: 1
+  },
+  navLinkText: {
+    color: color("blue", 50),
+    fontFamily: "brand-bold"
+  },
+  bottomNavContainerLink: {
+    flexDirection: "row",
+    alignItems: "center",
+    paddingHorizontal: size(4),
+    paddingVertical: size(1)
+  },
+  bottomNavContainerText: {
+    color: color("blue", 50),
+    fontSize: fontSize(-1)
+  }
+});
+
+interface BottomNavigationLink {
+  onPress: () => void;
+}
+
+export const BottomNavigationLink: FunctionComponent<BottomNavigationLink> = ({
+  children,
+  onPress
+}) => {
+  return (
+    <View style={styles.bottomNavContainerLink}>
+      <TouchableOpacity onPress={onPress}>
+        <AppText style={styles.bottomNavContainerText}>{children}</AppText>
+      </TouchableOpacity>
+    </View>
+  );
+};
+
+export const DrawerButton: FunctionComponent<DrawerButtons> = ({
+  icon,
+  label,
+  onPress
+}) => {
+  return (
+    <TouchableOpacity onPress={onPress}>
+      <View style={styles.navLink}>
+        <MaterialCommunityIcons
+          name={icon}
+          size={size(2.5)}
+          color={color("blue", 50)}
+          style={{ marginRight: size(2) }}
+        />
+        <AppText style={styles.navLinkText}>{label}</AppText>
+      </View>
+    </TouchableOpacity>
+  );
+};
+
+export const DrawerNavigationComponent: FunctionComponent<DrawerContentComponentProps> = ({
+  navigation
+}) => {
+  const { logout } = useLogout();
+  const showHelpModal = useContext(HelpModalContext);
+  const { drawerButtons } = useDrawerContext();
+  const handleLogout = useCallback((): void => {
+    logout(navigation.dispatch);
+  }, [logout, navigation.dispatch]);
+
+  const onPressLogout = (): void => {
+    Alert.alert(
+      "Are you sure you want to log out?",
+      "Stay logged in to change your scanning location.",
+      [
+        {
+          text: "Cancel"
+        },
+        {
+          text: "Logout",
+          onPress: handleLogout,
+          style: "destructive"
+        }
+      ],
+      { cancelable: false }
+    );
+  };
+
+  const onPressCloseDrawer = (): void => {
+    navigation.dispatch(DrawerActions.closeDrawer());
+  };
+
+  return (
+    <SafeAreaView
+      style={styles.container}
+      forceInset={{ top: "always", horizontal: "never" }}
```

Comment:
What's the purpose of this prop?

Reply:
This is to ensure that the links don't get block by the notch at the top:
https://reactnavigation.org/docs/3.x/handling-iphonex/#use-forceinset-to-get-more-control

Reply:
By default doesn't the top inset always exist? Do we need the horizontal one given that the app only works in portrait?